ext/intl/timezone/timezone_methods.cpp

```diff
@@ -133,53 +133,45 @@ U_CFUNC PHP_FUNCTION(intltz_get_unknown)
 U_CFUNC PHP_FUNCTION(intltz_create_enumeration)
 {
 	zval				*arg = NULL;
-	StringEnumeration	*se	  = NULL;
+	zend_string                     *args = NULL;
+	zend_long                       argl = 0;
+	StringEnumeration	        *se = NULL;
 	intl_error_reset(NULL);
 
-	/* double indirection to have the zend engine destroy the new zval that
-	 * results from separation */
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "|z", &arg) == FAILURE) {
-		RETURN_THROWS();
-	}
+	ZEND_PARSE_PARAMETERS_START(0, 1)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_STR_OR_LONG(args, argl);
+	ZEND_PARSE_PARAMETERS_END();
 
-	if (arg == NULL || Z_TYPE_P(arg) == IS_NULL) {
+	if (ZEND_NUM_ARGS() == 0) {
 		se = TimeZone::createEnumeration();
```

Comment:
Oh maybe the first arg should then be ``string|int|null``, sorry didn't realize this at first.

Reply:
@kocsismate do you think the param should be nullable or not? Or is that not really an issue in regards to the RFC about overloaded methods?

Reply:
Uh, thanks for spotting this. The argument should be definitely nullable, otherwise it will be overloaded. @devnexen In most of the cases, using `ZEND_NUM_ARGS()` is a clear sign of an overloaded signature.